internal/bcdb/db.go

```diff
@@ -395,6 +418,45 @@ func (d *db) GetData(dbName, querierUserID, key string) (*types.GetDataResponseE
 	}, nil
 }
 
+// DataQuery executes a given JSON query and return key-value pairs which are matching
+// the criteria provided in the query. The query is a json marshled bytes which needs
+// to contain a top level combinational operator followed by a list of attributes and
+// a list of conditions per attributes. For example, the following is one of the query:
+//
+// {
+//   "selector": {
+// 		"$and": {            -- top level combinational operator
+// 			"attr1": {          -- a field in the json document
+// 				"$gte": "a",    -- value criteria for the field
+// 				"$lt": "b"      -- value criteria for the field
+// 			},
+// 			"attr2": {          -- a field in the json document
+// 				"$eq": true     -- value criteria for the field
+// 			},
+// 			"attr3": {          -- a field in the json document
+// 				"$lt": "a2"     -- a field in the json document
+// 			}
+// 		}
+//   }
+// }
```

Comment:
I don't think you need to put the same comments twice, either on the interface or here is fine, I think